ledger/participant-integration-api/src/main/scala/platform/indexer/MeteringAggregator.scala

```diff
@@ -0,0 +1,176 @@
+      val startUtcTime: OffsetDateTime = toOffsetDateTime(lastLedgerMeteringEnd.timestamp)
+      val endUtcTime = startUtcTime.plusHours(1)
+
+      if (nowUtcTime.isAfter(endUtcTime)) {
+
+        val endTime = toTimestamp(endUtcTime)
+        val ingestedLedgerEnd = parameterStore.ledgerEnd(conn).lastOffset
+        val maybeMaxOffset =
+          meteringStore.transactionMeteringMaxOffset(lastLedgerMeteringEnd.offset, endTime)(conn)
+
+        val (periodIngested, hasMetering, thisOffsetEnd) = maybeMaxOffset match {
+          case Some(offset) => (offset <= ingestedLedgerEnd, true, offset)
+          case None => (true, false, lastLedgerMeteringEnd.offset)
+        }
+
+        if (periodIngested) {
+          Some(
+            aggregate(
+              conn = conn,
+              lastLedgerMeteringEnd = lastLedgerMeteringEnd,
+              thisLedgerMeteringEnd = LedgerMeteringEnd(thisOffsetEnd, endTime),
+              hasMetering = hasMetering,
+            )
+          )
+        } else {
+          logger.info("Not all transaction metering for aggregation time period is yet ingested")
+          None
+        }
+      } else {
+        None
+      }
+    }
+
+    future.onComplete({
+      case Success(None) => logger.info("No transaction metering aggregation required")
+      case Success(Some(lme)) =>
+        logger.info(s"Aggregating transaction metering completed up to $lme")
+      case Failure(e) => logger.error("Failed to aggregate transaction metering", e)
+    })
+
+    future.map(_ => ())
+  }
+
+  private def aggregate(
+      conn: Connection,
+      lastLedgerMeteringEnd: LedgerMeteringEnd,
+      thisLedgerMeteringEnd: LedgerMeteringEnd,
+      hasMetering: Boolean,
+  ): LedgerMeteringEnd = {
+    logger.info(s"Aggregating transaction metering for $thisLedgerMeteringEnd")
+
+    if (hasMetering) {
+      populateParticipantMetering(conn, lastLedgerMeteringEnd, thisLedgerMeteringEnd)
+    }
+
+    parameterStore.updateLedgerMeteringEnd(thisLedgerMeteringEnd)(conn)
+
+    thisLedgerMeteringEnd
+  }
+
+  private def getLastOrInitializeLedgerMeteringEnd(
+      conn: Connection,
+      nowUtcTime: OffsetDateTime,
+  ): LedgerMeteringEnd = {
+    parameterStore.ledgerMeteringEnd(conn).getOrElse {
+      val initUtcTimestamp = meteringStore
+        .transactionMeteringMinTimestamp(conn)
+        .map(t => toOffsetDateTime(t))
+        .getOrElse(nowUtcTime)
+        .truncatedTo(ChronoUnit.HOURS)
+        .minusHours(1)
+      LedgerMeteringEnd(Offset.beforeBegin, toTimestamp(initUtcTimestamp))
+    }
+  }
+
+  private def populateParticipantMetering(
+      conn: Connection,
+      lastLme: LedgerMeteringEnd,
+      thisLme: LedgerMeteringEnd,
+  ): Unit = {
+
+    val transactionMetering: Seq[MeteringStore.TransactionMetering] =
+      meteringStore.transactionMetering(lastLme.offset, thisLme.offset)(conn)
+
+    val participantMetering = transactionMetering
```

Comment:
@nmarton-da - I had a thought that this could all be done more efficiently by the DB.  Let's discuss later.